llvm_util/llvm2alive.cpp

```diff
@@ -692,15 +692,29 @@ class llvm2alive_ : public llvm::InstVisitor<llvm2alive_, unique_ptr<Instr>> {
 
   RetTy visitUnreachableInst(llvm::UnreachableInst &i) {
     auto fals = get_operand(llvm::ConstantInt::getFalse(i.getContext()));
-    return make_unique<Assume>(*fals, /*if_non_poison=*/false);
+    return make_unique<Assume>(*fals, Assume::AndNonPoison);
   }
 
   RetTy visitIntrinsicInst(llvm::IntrinsicInst &i) {
     switch (i.getIntrinsicID()) {
     case llvm::Intrinsic::assume:
     {
+      unsigned n = i.getNumOperandBundles();
+      for (unsigned j = 0; j < n; ++j) {
+        auto bundle = i.getOperandBundleAt(j);
+        llvm::StringRef name = bundle.getTagName();
+        if (name == "noundef") {
+          // Currently it is unclear whether noundef bundle takes single op.
+          // Be conservative & simply assume it can take arbitrary num of ops.
+          for (unsigned j = 0; j < bundle.Inputs.size(); ++j) {
+            llvm::Value *v = bundle.Inputs[j].get();
+            BB->addInstr(
+              make_unique<Assume>(*get_operand(v), Assume::WellDefined));
```

Comment:
what if get_operand returns null?

Reply:
I think it should be non-null, because def should dominate operand bundles: https://godbolt.org/z/xqE4Tb
It is assumed as non-null in other places, e.g., `visitUnreachableInst`.